plugins/filter_lua/CMakeLists.txt

```diff
@@ -7,3 +7,7 @@ if(MSVC)
 else()
   FLB_PLUGIN(filter_lua "${src}" "m")
 endif()
+
+if(FLB_MPACK)
```

Comment:
since mpack is always included and we want to give the option to choose if Lua Filter will use or not mpack, we can make the option as `FLB_FILTER_USE_MPACK` default to `yes` , and if set, then uses mpack backend. basically rename `FLB_MPACK` to `FLB_FILTER_LUA_USE_MPACK`